puppet/zulip/manifests/postgresql_base.pp

```diff
@@ -83,28 +83,20 @@
     }
 
     package{"${postgresql}-pgdg-pgroonga":
-      ensure  => installed,
-      require => [Package[$postgresql],
-                  Exec[$setup_system_deps]],
+      ensure  => latest,
+      require => [
+        Package[$postgresql],
+        Exec[$setup_system_deps]
+      ],
     }
-
-    $dbname = zulipconf('postgresql', 'database_name', 'zulip')
-    $dbuser = zulipconf('postgresql', 'database_user', 'zulip')
-    file { $pgroonga_setup_sql_path:
-      ensure  => file,
+    exec { 'pgroonga-config':
       require => Package["${postgresql}-pgdg-pgroonga"],
-      owner   => 'postgres',
-      group   => 'postgres',
-      mode    => '0640',
-      content => template('zulip/postgresql/pgroonga_setup.sql.template.erb'),
-    }
-
-    exec{'create_pgroonga_extension':
-      require => File[$pgroonga_setup_sql_path],
-      # lint:ignore:140chars
-      command => "bash -c 'cat ${pgroonga_setup_sql_path} | su postgres -c \"psql -v ON_ERROR_STOP=1 ${dbname}\" && touch ${pgroonga_setup_sql_path}.applied'",
-      # lint:endignore
-      creates => "${pgroonga_setup_sql_path}.applied",
+      unless  => @("EOT"/$),
+          test -f ${pgroonga_setup_sql_path}.applied &&
+          test "$(dpkg-query --show --showformat='\${Version}' "${postgresql}-pgdg-pgroonga")" \
+             = "$(cat ${pgroonga_setup_sql_path}.applied)"
+          | EOT
```

Comment:
It seems that this is redundant because `pgroonga-config` also has the same check.

Reply:
I put it in both places defensively -- doing it here is more important, though, as it keeps the puppet application a no-op if the updated version has already been applied.  I dropped the check in the shell script, since it's a safe no-of if the version matches already.